src/main/java/seedu/address/model/person/TagContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,47 @@
+package seedu.address.model.person;
+
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Tests that a {@code Person} is tagged with the corresponding tag.
+ */
+public class TagContainsKeywordsPredicate implements Predicate<Person> {
+    private final String tag;
```

Comment:
Should the type of the variable be Tag instead? Else maybe the variable can be named tagName instead, for clarity. The name of the tag can be accessed via getTagName().

Reply:
```suggestion
    private final Tag tag;
```